Enhance / refactor some test-suite code

AndrejMitrovic opened the pull request:
Split off from https://github.com/bpfkorea/agora/pull/109

AndrejMitrovic committed:
Move getAllBocks() / containSameBlocks() to Base.d

These will be used by multiple unittest modules.

AndrejMitrovic committed:
Change makeTestNetwork() to be more configurable

The retry_delay / max_retries / timeout are now configurable,
making it easier to write specialized tests (for example to
make the Thread sleep less often between requests).

Geod24 commented on source/agora/test/Base.d:
```diff
@@ -303,14 +303,17 @@ public enum NetworkTopology
 public NetworkT makeTestNetwork (NetworkT : TestNetworkManager)
-    (NetworkTopology topology, size_t nodes, bool configure_network = true)
+    (NetworkTopology topology, size_t nodes, bool configure_network = true,
+        long retry_delay = 100, size_t max_retries = 20, long timeout = 500)
```
What are the units of `100` and `500`?

AndrejMitrovic replied:
I'll document it.

Geod24 commented:
Grey :(

AndrejMitrovic committed:
Change and rename tryUntil() function

It's more intuitive to pass the condition first,
and then the timeout second.

Renamed it to retryFor() and swapped order of params.

AndrejMitrovic commented:
Rebased, and added a commit implementing the suggestion at https://github.com/bpfkorea/agora/pull/144#issuecomment-518587806

AndrejMitrovic commented:
> Grey :(

Green.

Geod24 reviewed (approved):
Re second commit: Looks like we will end up just passing a `Config` struct.
